packages/compiler-vapor/src/transforms/vIf.ts

```diff
@@ -52,37 +54,109 @@ export function processIf(
         positive: branch,
       })
     }
+  } else {
+    // check the adjacent v-if
+    const siblings = context.parent!.node.children
+    const comments = []
+    let sibling: TemplateChildNode | undefined
+    let i = siblings.indexOf(node)
+    while (i-- >= -1) {
+      sibling = siblings[i]
+
+      if (sibling && sibling.type === NodeTypes.COMMENT) {
+        context.removeNode(sibling)
```

Comment:
AST should not be mutated. Unlike old compiler, we only need to generate IR node, and keep AST not changed.